src/compute/aggregate/sum.rs

```diff
@@ -23,19 +23,20 @@ pub trait Sum<T> {
 #[clone(target = "x86_64+avx")]
 fn nonnull_sum<T>(values: &[T]) -> T
 where
-    T: NativeType + Simd,
-    T::Simd: Add<Output = T::Simd> + Sum<T>,
+    T: NativeType + Simd + Add<Output = T> + std::iter::Sum<T>,
+    T::Simd: Sum<T> + Add<Output = T::Simd>,
 {
-    let mut chunks = values.chunks_exact(T::Simd::LANES);
-
-    let sum = chunks.by_ref().fold(T::Simd::default(), |acc, chunk| {
-        acc + T::Simd::from_chunk(chunk)
-    });
-
-    let remainder = T::Simd::from_incomplete_chunk(chunks.remainder(), T::default());
-    let reduced = sum + remainder;
+    // Safety:
+    // T::Simd is the vector type T and the alignment is similar to aligning to [T; alignment]
```

Comment:
I think that we need to make the trait `NativeSimd` `unsafe` and describe the invariant that it must uphold (that alignments match).

I would also move this to a (`safe`) inlined function in the trait `Simd`, so that the `unsafe` is close to where the contract is defined (as it may be used in other instances)

Reply:
Good idea.